lib/jsonapi/rspec/attributes.rb

```diff
@@ -1,26 +1,27 @@
+require 'pry'
 module JSONAPI
   module RSpec
     module Attributes
       ::RSpec::Matchers.define :have_attribute do |attr|
         match do |actual|
-          (actual['attributes'] || {}).key?(attr.to_s) &&
-            (!@val_set || actual['attributes'][attr.to_s] == @val)
+          actual = JSONAPI::RSpec.as_indifferent_hash(actual)
+          val = (actual || {}).dig('attributes', attr.to_s)
+          !val.nil? && (!@with_value || val == @val)
         end
 
         chain :with_value do |val|
-          @val_set = true
+          @with_value = true
```

Comment:
In hindsight - definitely should be in its own "variable naming consistency" type PR.  I'll revert these but I think it would be good to standardise on a naming format for the chain booleans.  One is the method name and the others are [something]_set. 

@val_set isn't very clear (